g1ant-language/language-basics.md

```diff
@@ -0,0 +1,321 @@
+
+A command can be simple, one-word, like `dialog` or extended, like `selenium.open`. Two-word commands separated with a dot in the middle belong to a family of commands. It means that there are other commands starting with `selenium.` and their suffix (the part following the dot) expresses certain activity: `open`, `newtab` ,`switch`.
+
+Each family of commands has its special ‘activities’. Once you start typing the command inside the script editor window, a list of available activities will pop up, as you can see below:
+
+![](.gitbook/assets/selenium.png)
+
+### Argument
+
+A command needs to know, what exactly it is supposed to do. So, you must provide some information — and this data is called an *argument*. For example, when you type `program`, the robot will suggest you next input in a drop-down list and the first item will be `name`. It is an argument for the `program` command followed with a name of the program to be launched. So, if you wanted the robot to open Notepad, you would write `program name notepad` — remember the “command argument value” rule?
+
+### Value
+
+The content of a field or a variable is called a *value*. It can refer to alphabetic as well as numeric data. For example, in the expression
+
+```G1ANT
+selenium.open type ‴firefox‴ url ‴google.com‴ result ♥google
+```
+
+the elements `‴firefox‴` and `‴google.com‴` are values. Note that text values are embraced with triple prime characters available by pressing **Ctrl+'** (apostrophe) keyboard shortcut or by clicking the text symbol icon on the toolbar.
+
+Values always belong to one type of structures, [variables](), and they can be any text, numbers and a mixture of both (URL address, for instance). Values are strictly related to their arguments.
+
+### Examples
+
+In the following example the golden “command-argument-value” rule is a bit twisted:
+
+```G1ANT
+text.read c:\Users\♥USERNAME\Desktop\g1ant.txt result ♥g1
+```
+
+The `text.read` command is immediately followed by a value (`c:\Users\♥USERNAME\Desktop\g1ant.txt`), not an argument. It’s because some argument names can be omitted, since they are obvious and mandatory. In this case, the missing argument name is `filename`, which is followed by a value with the name of a file to open. Then, there is the `result` argument and its `♥g1` value.
+
+Other examples of arguments that can be skipped and replaced by their values, include `name` for the `program` command (which specifies the program to be launched) or `title` for the `window`command (which specifies the title of the window to focus on):
+
+`program notepad` instead of `program name notepad`
+
+`window *chrome` instead of `window title *chrome`
+
+Note the asterisk before `chrome`: it serves as a wildcard for all characters preceding the word “chrome”, so that any Google Chrome browser window would be selected no matter what its full title is.
```

Comment:
```suggestion
Note the special asterisk symbol before `chrome`: it's called Search Place and serves as a wildcard for all characters preceding the word “chrome”, so that any Google Chrome browser window would be selected no matter what its full title is. You can enter this character from `Insert/Search Place` menu, with **Ctrl+8** keyboard shortcut or by clicking the ✱ icon on the toolbar.
```